content/crutchcorn/collections/art-of-accessibility/posts/art-of-a11y-text/index.md

```diff
@@ -0,0 +1,561 @@
+
+This mockup might be naïvely marked up as such:
+
+```html
+<h1>EchoFlow X1</h1>
+<img src="headphone.jpg" alt="Black headphones with 'EchoFlow X' written on the earcup.">
+<h2 class="visually-hidden">Description</h2>
+<p>The EchoFlow X1 is the world's first...</p>
+<h2>Features</h2>
+<ul>
+    <li>Infinite Playback</li>
+    <li>Adaptive Sound</li>
+</ul>
+<h2>Customer Reviews</h2>
+<p>Mind blowing! It's like...</p>
+<p>A game-changer for music...</p>
+```
+
+But this introduces a problem: Where does one part of the app stop and another part begin?
+
+This is where `<section>` tags come into play. Let's change our markup to the following:
+
+```html
+<h1>EchoFlow X1</h1>
+<main>
+  <img src="headphone.jpg" alt="Black headphones with 'EchoFlow X' written on the earcup.">
+
+  <section>
+    <h2 class="visually-hidden">Description</h2>
+    <p>The EchoFlow X1 is the world's first...</p>
+  </section>
+
+  <section>
+    <h2>Features</h2>
+    <ul>
+        <li>Infinite Playback</li>
+        <li>Adaptive Sound</li>
+  	</ul>
+  </section>
+
+  <section>
+    <h2>Customer Reviews</h2>
+    <ul>
+        <li>Mind blowing! It's like...</li>
+        <li>A game-changer for music...</li>
+      </ul>
+  </section>
+</main>
+```
+
+Now it's more clear which headings belong to which sections.
+
+>  **Keep in mind:**
+>
+> Sections should almost always have a heading as the first child; otherwise it's challenging to figure out what a given section is intended to do.
+
+# Invisible Text
+
+In our previous code sample, you may have noticed that we had the following:
+
+```html
+<h2 class="visually-hidden">Description</h2>
+```
+
+> What is that class doing?
+
+Well, as the name suggests, `visually-hidden` is a custom class that allows you to keep text accessible by screen-readers and programmatic access, but hidden visually:
+
+```css
+.visually-hidden {
+  position: absolute;
+  width: 1px;
+  height: 1px;
+  padding: 0;
+  margin: -1px;
+  overflow: hidden;
+  clip: rect(0, 0, 0, 0);
+  white-space: nowrap;
+  border: 0;
+}
+```
+
+> Why would we want to have a heading that's visually hidden?
+
+Good question! In the example we showed earlier, it allows for further clarity in the `<section>` to have an associated header.
+
+Let's take a look at some common "do" and "do not"s with visually hidden text.
+
+**Do**:
+
+- Use invisible text to convey information to the user that might only be visual otherwise:
+
+    ```html
+    <span>
+        <span aria-hidden="true">★★★★☆</span>
+        <span class="visually-hidden"> Rating: 4 out of 5 stars</span>
+    </span>
+    ```
+
+    ```html
+    <button>
+        <span class="visually-hidden">Settings</span>
+        <svg aria-hidden="true">...</svg>
+    </button>
+    ```
+
+    > **Tip:**
+    >
+    > You may even want to add in a tooltip when the user is focused or hovered on a visual-only element to display this text as well.
```

Comment:
I am not sure about this being recommending alongside, since a visual-only (`aria-hidden="true"`) element in the tab order would result in a barrier for assistive technology users, and not entering it into the tab order would create a barrier for keyboard users.

And it still leaves many UX issues from tooltips left to the user to solve like mobile users and affordance.

If it's worth abbreviating for a non-disabled user, do the same for a disabled user and don't confuse them or increase their verbosity. If it's worth adding a tooltip to explain it to a non-disabled user, do it for the disabled user.

Reply:
This recommendation isn't about text abbreviation, it's about things like icon-only elements like a sidebar or bottom nav, et al.

I would argue that it's part of good UX to provide tooltips/hover effects to explain these kinds of things